streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/ProcessorParameters.java

```diff
@@ -114,13 +114,12 @@ KTableSource<KIn, VIn> kTableSourceSupplier() {
 
     @SuppressWarnings("unchecked")
     <KR, VR> KTableProcessorSupplier<KIn, VIn, KR, VR> kTableProcessorSupplier() {
-        // This cast always works because KTableProcessorSupplier hasn't been converted yet.
```

Comment:
Now that TableProcessor suplier is using latest API, I think this is worth updating

Reply:
What happens if/when we call this and it returns `null`? Is that pretty much guaranteed to be a bug/result in an NPE, or are there valid uses for returning null? Sorry, wasn't keeping up with these changes so just trying to wrap my head around it

Reply:
> Is that pretty much guaranteed to be a bug/result in an NPE, or are there valid uses for returning null
Looking at the uses of these methods, it could be both. There are those that expect a value and will crash with NPE (bug) (e.g. TableSourceNode), and there are others that handle null values and do something else (e.g. TableProcessorNode).

Considering the usage of these methods, we could consider removing them and handle the casting on the classes using it. I have created https://issues.apache.org/jira/browse/KAFKA-14409 to discuss this.